geth/jail/jail_cell.go

```diff
@@ -65,6 +63,12 @@ func (cell *JCell) Fetch(url string, callback func(otto.Value)) (otto.Value, err
 			});
 		});
 	`)
+
+	if err != nil {
+		return val, err
+	}
+
+	return val, cell.lo.Run()
```

Comment:
This must be called here to ensure fetch executes immediately in the event loop.